### PR TITLE
Scattergeo and choropleth click and hover events

### PR DESCRIPTION
**Pull request by etpinard**
@mdtusz @cpsievert 

About time I do this.

### CODE REVIEW

**Comment by cpsievert** [comment]
:boom: :tada:

File: src/traces/scattergeo/plot.js [hunk 1]
```diff
@@ -116,19 +116,14 @@ function makeLineGeoJSON(trace) {
...
+        .selectAll('g.trace.scattergeo')
         .data(scattergeoData);
 
     gScatterGeoTraces.enter().append('g')
-            .attr('class', 'trace scattergeo');
+        .attr('class', 'trace scattergeo');
+
+    gScatterGeoTraces.exit().remove();
 
     // TODO add hover - how?
```
Comment (mdtusz) [inline src/traces/scattergeo/plot.js]: Don't need this anymore :tada:

Comment (etpinard) [inline src/traces/scattergeo/plot.js]: oh, we still need this unfortunately. scattergeo trace with mode `'lines'` don't have any hover text at the moment.

File: src/traces/choropleth/plot.js [hunk 2]
```diff
@@ -59,46 +59,49 @@ plotChoropleth.calcGeoJSON = function(trace, topojson) {
...
                     y: xy[1],
-                    name: d.nameLabel,
-                    text: d.textLabel
+                    name: pt.nameLabel,
+                    text: pt.textLabel
                 }, {
                     container: geo.hoverContainer.node()
                 });
+
+                geo.graphDiv.emit('plotly_hover', eventDataFunc(pt, ptIndex));
```
Comment (mdtusz) [inline src/traces/choropleth/plot.js]: :tada:

File: src/traces/choropleth/plot.js [hunk 3]
```diff
@@ -59,46 +59,49 @@ plotChoropleth.calcGeoJSON = function(trace, topojson) {
...
+            }
+
+            function handleClick(pt, ptIndex) {
+                geo.graphDiv.emit('plotly_click', eventDataFunc(pt, ptIndex));
```
Comment (mdtusz) [inline src/traces/choropleth/plot.js]: :tada:

File: test/jasmine/tests/geo_interact_test.js [hunk 4]
```diff
@@ -14,14 +14,24 @@ describe('Test geo interactions', function() {
 
     describe('mock geo_first.json', function() {
         var mock = require('@mocks/geo_first.json');
+        var gd;
+
+        function mouseEventScatterGeo(type) {
+            mouseEvent(type, 300, 235);
```
Comment (mdtusz) [inline test/jasmine/tests/geo_interact_test.js]: It still surprises me that this isn't a more widely found solution when searching for "mouse event testing".

**Comment by mdtusz** [comment]
Nice work. 
:dancer:

**Pull request merged**